### PR TITLE
Adding type customer b2b corporate to type employee and instant

### PR DESCRIPTION
**Pull request by wahyu-di**
DEV10134 - Adding type customer b2b corporate to type employee and instant

### CODE REVIEW

**Comment by wahyu-di** [comment]
DEV10134 - Adding type customer b2b corporate to type employee and instant

**Review by Rizki-H — CHANGES_REQUESTED** [review]
Please update the local branch first

File: Production/customer_invoice_raw_part_1.sql [hunk 1]
```diff
@@ -170,6 +202,7 @@ fd as (
     , reseller_id
     , cc_installment 
     , total_customer_price
+    , split(from_email,'@') [offset(1)] as email_domain
```
Comment (Rizki-H) [inline Production/customer_invoice_raw_part_1.sql:205]: Please use safe_offset

File: Production/customer_invoice_raw_part_2.sql [hunk 2]
```diff
@@ -1495,8 +1506,10 @@ group by 1
         then 'GTN_SGP'
       else 'GTN_IDN' end as company
     , case
-          when date(payment_timestamp) >= '2020-04-06' and o.corporate_flag is not null and o.payment_source  in ('cash_onsite','corporate_deposit','corporate_invoice','corporate_billing') then o.business_id /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/
-          when date(payment_timestamp) <'2020-04-06' and o.corporate_flag is not null then o.business_id /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/ /*2021=07-18 - adding new payment source for b2b corporate*/
+          when date(payment_timestamp) >= '2020-04-06' and o.corporate_flag = 'invoice' and o.payment_source  in ('cash_onsite','corporate_deposit','corporate_invoice','corporate_billing') then o.business_id /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/
+          -- when date(payment_timestamp) <'2020-04-06' and o.corporate_flag is not null then o.business_id /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/ /*2021=07-18 - adding new payment source for b2b corporate*/
```
Comment (Rizki-H) [inline Production/customer_invoice_raw_part_2.sql:1510]: Please use slash (/) & star (*) as a comment

File: Production/customer_invoice_raw_part_2.sql [hunk 3]
```diff
@@ -1513,8 +1526,10 @@ group by 1
           when o.reseller_type in ('reseller','widget') then safe_cast(o.reseller_id as string)
         end as customer_id
     , case
-          when date(payment_timestamp) >= '2020-04-06' and o.corporate_flag is not null and o.payment_source  in ('cash_onsite','corporate_deposit','corporate_invoice','corporate_billing') then 'B2B Corporate' /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/
-          when date(payment_timestamp) <'2020-04-06' and o.corporate_flag is not null then 'B2B Corporate' /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/
+          when date(payment_timestamp) >= '2020-04-06' and o.corporate_flag = 'invoice' and o.payment_source  in ('cash_onsite','corporate_deposit','corporate_invoice','corporate_billing') then 'B2B Corporate' /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/
+          -- when date(payment_timestamp) <'2020-04-06' and o.corporate_flag is not null then 'B2B Corporate' /* 2020 04 06 - additional request for B2B Corporate only payment source cash_onsite*/
```
Comment (Rizki-H) [inline Production/customer_invoice_raw_part_2.sql:1530]: Please use slash (/) & star (*) as a comment

File: Production/supplier_invoice_raw_part_1.sql [hunk 4]
```diff
@@ -1578,27 +1591,31 @@ fd as (
...
+    --         and (string_agg(distinct corporate_flag) is null or (string_agg(distinct corporate_flag) is not null and string_agg(distinct payment_source) not in ('cash_onsite','corporate_deposit','corporate_invoice','corporate_billing')))
+    --           then 'C-000001'
+    --       when 
+    --         date(max(payment_timestamp_oc)) < '2020-04-06' and string_agg(distinct reseller_type) in ('none','online_marketing','native_apps')
+    --         and (string_agg(distinct corporate_flag) is null)
+    --           then 'C-000001'
+    --       when string_agg(distinct reseller_type) in ('none','online_marketing','native_apps') then 'C-000001'
+    --       when string_agg(distinct reseller_type) in ('tiket_agent','txtravel','agent','affiliate') then string_agg(distinct safe_cast(reseller_id as string))
+    --       when string_agg(distinct reseller_type) in ('reseller','widget') then string_agg(distinct safe_cast(reseller_id as string))
+    --     end as customer_reference_id
```
Comment (Rizki-H) [inline Production/supplier_invoice_raw_part_1.sql:1618]: Please use slash (/) & star (*) as a comment

**Review by Rizki-H — CHANGES_REQUESTED** [review]
Need some minor changes

**Review by Rizki-H — APPROVED** [review]
LGTM

**Comment by anggianggara** [comment]
approved

**Pull request merged**